### ISSUE TITLE
Implement CORS on all Vercel Apps

### ISSUE THREAD

**Issue by taariq** [VolumeFi/paloma#1101]
Implement CORS on Vercel as upstream data providers require that the header be respected:

> We provide an Article on how to enable CORS on your Serverless Functions: 
https://vercel.com/guides/how-to-enable-cors

NOTE: **This also applies if you're trying to access a resource from an upstream provider** We need to implement CORS because we are attempting to access the RPC resources from bloxroute.

**Comment by taariq** [comment]
The CORS error: 
Access to fetch at 'https://volume.liquify.com/api=0440032/eth' from origin 'https://www.curvebot.fi' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by taariq** [comment]
Fixed with:
https://github.com/VolumeFi/curvefi-bot/pull/8
https://github.com/VolumeFi/palomabot/pull/297

**Comment by verabehr** [comment]
I'm still getting CORS errors as well. This is also blocking the UI from displaying the max borrow amount which makes the screen somewhat unusable. 
<img width="1429" alt="Screen Shot 2023-12-18 at 1 55 25 PM" src="https://github.com/VolumeFi/paloma/assets/19958639/7e6d4893-0666-46ca-a54c-f3325a27a9d0">

```
_app-c9caece94044acfb.js:1 
        
        
       POST https://eth-protect.rpc.blxrbdn.com/ net::ERR_FAILED 429 (Too Many Requests)
o @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
a @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
d @ _app-c9caece94044acfb.js:1
h @ _app-c9caece94044acfb.js:1
send @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
s @ _app-c9caece94044acfb.js:1
Promise.then (async)
c @ _app-c9caece94044acfb.js:1
o @ _app-c9caece94044acfb.js:1
Promise.then (async)
c @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
f @ _app-c9caece94044acfb.js:1
_uncachedDetectNetwork @ _app-c9caece94044acfb.js:1
detectNetwork @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
s @ _app-c9caece94044acfb.js:1
Promise.then (async)
c @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
v @ _app-c9caece94044acfb.js:1
_ready @ _app-c9caece94044acfb.js:1
F @ _app-c9caece94044acfb.js:1
A @ _app-c9caece94044acfb.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
i @ 5392-62fb692d3744a9ec.js:1
e.init @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
a @ 5392-62fb692d3744a9ec.js:1
init @ 5392-62fb692d3744a9ec.js:1
e @ 1446-3987135f995413cb.js:1
(anonymous) @ 1446-3987135f995413cb.js:1
uI @ framework-a2363dd3c3dbe572.js:9
oU @ framework-a2363dd3c3dbe572.js:9
o @ framework-a2363dd3c3dbe572.js:9
x @ framework-a2363dd3c3dbe572.js:33
T @ framework-a2363dd3c3dbe572.js:33
1446-3987135f995413cb.js:1 Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.7.2)
    at f.makeError (_app-c9caece94044acfb.js:1:189313)
    at f.throwError (_app-c9caece94044acfb.js:1:189427)
    at A.<anonymous> (_app-c9caece94044acfb.js:1:270967)
    at Generator.throw (<anonymous>)
    at s (_app-c9caece94044acfb.js:1:262855)
curve-leverage-lending/:1 Access to fetch at 'https://eth-protect.rpc.blxrbdn.com/' from origin 'https://www.curvebot.fi' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
_app-c9caece94044acfb.js:1 
        
        
       POST https://eth-protect.rpc.blxrbdn.com/ net::ERR_FAILED
o @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
a @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
d @ _app-c9caece94044acfb.js:1
h @ _app-c9caece94044acfb.js:1
send @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
o @ _app-c9caece94044acfb.js:1
Promise.then (async)
c @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
f @ _app-c9caece94044acfb.js:1
_uncachedDetectNetwork @ _app-c9caece94044acfb.js:1
detectNetwork @ _app-c9caece94044acfb.js:1
(anonymous) @ _app-c9caece94044acfb.js:1
setTimeout (async)
(anonymous) @ _app-c9caece94044acfb.js:1
A @ _app-c9caece94044acfb.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
i @ 5392-62fb692d3744a9ec.js:1
e.init @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
(anonymous) @ 5392-62fb692d3744a9ec.js:1
a @ 5392-62fb692d3744a9ec.js:1
init @ 5392-62fb692d3744a9ec.js:1
e @ 1446-3987135f995413cb.js:1
(anonymous) @ 1446-3987135f995413cb.js:1
uI @ framework-a2363dd3c3dbe572.js:9
oU @ framework-a2363dd3c3dbe572.js:9
(anonymous) @ framework-a2363dd3c3dbe572.js:9
oI @ framework-a2363dd3c3dbe572.js:9
oC @ framework-a2363dd3c3dbe572.js:9
r4 @ framework-a2363dd3c3dbe572.js:9
(anonymous) @ framework-a2363dd3c3dbe572.js:9
setTimeout (async)
l @ _app-c9caece94044acfb.js:1068
(anonymous) @ _app-c9caece94044acfb.js:1068
(anonymous) @ _app-c9caece94044acfb.js:1068
(anonymous) @ _app-c9caece94044acfb.js:1
dispatch @ _app-c9caece94044acfb.js:1073
unsubscribe @ _app-c9caece94044acfb.js:1068
(anonymous) @ _app-c9caece94044acfb.js:1068
oP @ framework-a2363dd3c3dbe572.js:9
(anonymous) @ _app-c9caece94044acfb.js:1068
I @ 1377-8a77bdca41996044.js:1
(anonymous) @ 1377-8a77bdca41996044.js:1
E @ 1377-8a77bdca41996044.js:1
await in E (async)
e @ 1377-8a77bdca41996044.js:1
(anonymous) @ 1377-8a77bdca41996044.js:1
setTimeout (async)
(anonymous) @ 1377-8a77bdca41996044.js:1
uI @ framework-a2363dd3c3dbe572.js:9
oU @ framework-a2363dd3c3dbe572.js:9
(anonymous) @ framework-a2363dd3c3dbe572.js:9
oI @ framework-a2363dd3c3dbe572.js:9
oC @ framework-a2363dd3c3dbe572.js:9
r4 @ framework-a2363dd3c3dbe572.js:9
(anonymous) @ framework-a2363dd3c3dbe572.js:9
curve-leverage-lending/:1 Access to fetch at 'https://eth-protect.rpc.blxrbdn.com/' from origin 'https://www.curvebot.fi' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
_app-c9caece94044acfb.js:1 
        

    ```

**Comment by jamie088** [comment]
`Access to fetch at 'https://service.palomabot.ai/stats?' from origin 'https://www.curvebot.fi' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'https://curvebot.fi' that is not equal to the supplied origin. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

`Access to fetch at 'https://explorer-api.walletconnect.com/w3m/v1/getDesktopListings?projectId=b790153c46fe59c25875a033a6c378ed&sdkType=w3m&sdkVersion=js-2.7.1&page=1&entries=9&version=2' from origin 'https://www.curvebot.fi' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'https://curvebot.fi' that is not equal to the supplied origin. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by taariq** [comment]
Superseded by https://github.com/VolumeFi/paloma/issues/1109
